lib/pbench/server/utils.py

```diff
@@ -72,17 +73,18 @@ def quarantine(dest, logger, *files):
         sys.exit(101)
 
     for afile in files:
-        if not os.path.exists(afile) and not os.path.islink(afile):
+        afile = Path(afile)
+        if not afile.exists():
             continue
         try:
             # If the file we're moving is a tarball, update the dataset
             # state. (If it's the associated MD5 file, skip that.)
-            if str(afile).endswith(".tar.xz"):
+            if afile.name.endswith(".tar.xz"):
```

Comment:
We don't want to do this when the file is a symbolic link, right?

Reply:
Why? If it follows our normal naming conventions, this theoretically works fine even if we're seeing a broken "state" link.

The `except DatasetNotFound` protects against the consequences of trying to update a dataset that doesn't actually exist; but the suffix check keeps us from even trying if it's unlikely to represent a real dataset in the first place.

Reply:
Why would we want to mark a tar ball as quarantined, when there is a problem with a symlink?

I think this should be removed from this code path, and in all the places where `quarantine()` is called, have that invocation pass an additional argument to have the tar ball quarantined.

Reply:
After the prep-shim "pre-processing" phase, I'd expect any quarantine operation to be done using the state symlink. Ideally I'd like to capture more information about what happened in the `Dataset` object, but with this code structure we don't have any information beyond "the caller decided to quarantine". Yes, we could move it out a layer where we have more context, but that also increases the chances of missing a transition somewhere. Doing state transitions in common code is generally an advantage. This is all something to think about as we move away from "files" and "state links" to represent the data and context. For now, this feels like the best compromise.